foundation/blockchain/worker/sharetx.go

```diff
@@ -20,15 +25,30 @@ func (w *Worker) shareTxOperations() {
 	w.evHandler("worker: shareTxOperations: G started")
 	defer w.evHandler("worker: shareTxOperations: G completed")
 
-	for {
-		select {
-		case tx := <-w.txSharing:
-			if !w.isShutdown() {
-				w.state.NetSendTxToPeers(tx)
+	var wg sync.WaitGroup
```

Comment:
All of this is already running as a Goroutine. The Concurrency code is wrapped around this call. So no need for the extra concurrency here.